### PR TITLE
fix: use TS no-shadow rule

### PR DESCRIPTION
**Pull request by jquense**
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md

### CODE REVIEW

File: packages/eslint-config-4catalyzer-typescript/rules.js [hunk 1]
```diff
@@ -14,6 +14,15 @@ module.exports = {
   'import/default': 'off',
   'import/namespace': 'off',
 
+  'no-shadow': 'off',
+  '@typescript-eslint/no-shadow': [
+    'error',
+    {
+      ignoreTypeValueShadow: true,
+      ignoreFunctionTypeParameterNameValueShadow: true,
```
Comment (itajaja) [inline packages/eslint-config-4catalyzer-typescript/rules.js:22]: why would we ignore these? they both seem legit. I had to fix my code to comply but it made sense

Comment (jquense) [reply packages/eslint-config-4catalyzer-typescript/rules.js:22]: the first one we do intentionally in places and i didn't want to push another breaking change right now.
The function one seems actively wrong no? You want to be able to name function type parameters whatever is accurate, shadowing doesn't cause bugs in those cases. 

like why does it matter that `test` is used twice here? a consumer of the fn type isn't going to see anything else and giving it a worse name makes the type less clear to users of the type intellisense
```ts
const test = 1;
type Func = (test: string) => typeof test;
```

Comment (itajaja) [reply packages/eslint-config-4catalyzer-typescript/rules.js:22]: > why does it matter that `test` is used twice here?

for the same reason shadowing in general matters, I guess. I am fine if you are not a fan of this though!

Comment (jquense) [reply packages/eslint-config-4catalyzer-typescript/rules.js:22]: I think it's more convenient to allow it here! happy to revisit if it causes bugs

**Pull request merged**